test/unit/math/src/math.cpp

```diff
@@ -35,9 +35,19 @@ struct TestKernel
     //! @param acc Accelerator given from alpaka.
     //! @param functor Accessible with operator().
     ALPAKA_NO_HOST_ACC_WARNING
-    template<typename TAcc, typename TResults, typename TFunctor, typename TArgs>
-    ALPAKA_FN_ACC auto operator()(TAcc const& acc, TResults* results, TFunctor const& functor, TArgs const* args)
-        const noexcept -> void
+    template<
+        typename TAcc,
+        typename TMemoryHandleResults,
+        typename TMemoryHandleArgs,
+        typename TResults,
+        typename TIdx,
+        typename TFunctor,
+        typename TArgs>
+    ALPAKA_FN_ACC auto operator()(
+        TAcc const& acc,
+        alpaka::Accessor<TMemoryHandleResults, TResults, TIdx, 1, alpaka::WriteAccess> const results,
```

Comment:
Maybe another case for `BufferAccessor`?

Reply:
I think the code is good enough!